### ISSUE TITLE
feature: add `format_alt` and `format_list` to all blocks

### ISSUE THREAD

**Issue by martinpelikan** [greshake/i3status-rust#2014]
I looked briefly, so I might have missed it, but I can't find a way to add a `format_alt` to the `external_ip` block via the config. I suppose that means it needs to be added explicitly via the block code. That being said, I wonder if it would be more versatile to have add a `format_alt` in the core block code, and have a click handler to switch to it for all blocks, rather than only those that have it explicitly defined.

I understand other blocks such as `notify` have different behavior on left click, but I hope it should be possible to override the behavior in those or through the config without any conflict.

Additionally, it might also be nice to be able to use the scroll wheel to iterate through all the initialized (or configured) values available to the `format` code, and similarly override the scroll wheel behavior used in blocks like `volume`. So, for example, by default scrolling on `external_ip` would cycle through a `format_list` value containing: `["ip", "version", "city", "region", "region_code", "..."]` by default. The user could configure `format_list` to trim down to only those relevant to them, eg: 
```rust
[[block]]
block = "external_ip"
format = "$ip"
format_list = ["$ip", "$city", "$asn"]
```

P.S. Thank you for everyone's ongoing effort to improve the project. Every few months when I revisit my config and the release notes, there's always an improvement I appreciate.

**Comment by bim9262** [comment]
I don't think that we'd want to make every block have an alt format or list of formats. For example take a look at the pomodoro block which has several states in which clicking/scrolling have different actions.

**Comment by martinpelikan** [comment]
@bim9262 Thank you for the quick response.

Shall I change the request to only modify `external_ip` then? I suggested the change as a default because most blocks appear to follow the same pattern, while the interactive ones are more of an exception. Looking closer, there probably isn't much benefit, and might actually complicate existing code rather than simplify it.

**Comment by bim9262** [comment]
I'm sure some of the other non-interactive would benefit from having an alt format option. Instead of a format list option though mayt it would be better to allow a single or list of alt formats so that we don't need to set both.

See the Untagged section: https://serde.rs/enum-representations.html

The clock block may be interesting to you check out to see how we added the ability to display different timezones. While it's not actually a different format it's similar in concept.

**Comment by samuelshaaban** [comment]
This would be a good change. Doesn't have to be absolutely all blocks, sometimes exceptions make sense. However, there are still blocks that don't include this, most notably for me nvidia_gpu, which just completely prevents you from doing certain things. This to me is much more inconvenient than allowing a block to use it that may misbehave somewhat.

This should be changed from something like an opt-in system to an opt-out. The notable exceptions that might interfere with the ability to click like pomodoro or tea timer should be excepted, but all others should by default give the option.
